superset/views/core.py

```diff
@@ -282,7 +286,7 @@ def request_access(self) -> FlaskResponse:
         datasource_id = request.args.get("datasource_id")
         datasource_type = request.args.get("datasource_type")
         if datasource_id and datasource_type:
-            ds_class = ConnectorRegistry.sources.get(datasource_type)
+            ds_class = DatasourceDAO.sources.get(datasource_type)
```

Comment:
use `DatasourceType(datasource_type)` here

Reply:
DatasourceDAO.sources should be able to take either a string or the DatasourceType instance, so I think either works here.

Reply:
of course you'll get an error either way if the datasource_type is not valid.